src/DBConnection.hpp

```diff
@@ -1,14 +1,32 @@
 #pragma once
 
+#include <string>
+
+struct Info
+{
+    Info() : serverVersion(""), protocolVersion(0) {};
+    ~Info() = default;
+
+    std::string serverVersion;
+    int protocolVersion;
+};
+
 class IDBConnection
 {
 public:
-	virtual int query() const = 0;
+    virtual ~IDBConnection() = default;
+
+    virtual int query() const = 0;
 };
 
 class MySqlDBConnection : public IDBConnection
 {
 public:
-	int query() const override;
-	int advancedQuery() const;
+    explicit MySqlDBConnection(std::string serverVersion, int protocolVersion);
```

Comment:
@Morheit , why ctor is explicit?

Reply:
@malirod to prevent availability for implicit conversion or in other words for such code: `MySqlDBConnection connection = {"Example", 10};`
Personal opinion that such code isn't "beautiful"

Reply:
@Morheit  , are you sure that implicit conversion is valid for the ctor with more that one param?

Reply:
@malirod, yes. In C++11 multi-parameter constructors can be implicitly converted to with brace initialization. However, before C++11 explicit only applied to single-argument constructors. For multiple-argument constructors, it was ignored and had no effect.